wireguard-go/README.md

```diff
@@ -0,0 +1,43 @@
+This directory contains to patches that can be applied to wireguard-go to leverage wolfCrypt as a crypto engine. This can run on a Linux or macOS machine and requires a Go version > 1.20.
+
+## Patches
+- `Wireguard-Go-wolfCrypt-Port.patch` : This patch simply ports the cryptography to use the go-wolfssl wrapper and wolfSSL. The underlying algorithms used in wireguard are the same.
+- `Wireguard-Go-FIPS-wolfCrypt-port.patch` : This patch ports wireguard to use FIPS 140-3 wolfCrypt and FIPS approved algorithms. Blake2s is replaced with SHA-256, Chaha20_Poly1305 with AES-GCM, and X25519 with ECC P-256.
+
+## Building
+First, build and install both wolfSSL and go-wolfssl following the instructions here: https://github.com/wolfSSL/go-wolfssl/blob/master/README.md. Make sure that wolfSSL is configured as shown below for the non-FIPS build.
+```
+./configure --enable-blake2s --enable-xchacha --enable-curve25519
+```
+
+If you're interested in the FIPS build, configure as shown below. Make sure that wolfCrypt tests pass, the normal flow for doing this can be seen below.
```

Comment:
Maybe add some clarification that these interactions are done in `XXX-fips-test`.

Reply:
Done